### PR TITLE
GTEST/UCP/SOCKADDR: randomize iface selection for testing

### PR DESCRIPTION
**Pull request by evgeny-leksikov**
## What
select random interface for each test, important if there are several
differently configured interfaces, for example IPoIB, RoCE, IPv4/6

## Why ?
to improve test coverage

### CODE REVIEW

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>FAILED</b> on 5 of 17 workers (click for details)</summary>

**Note:** the logs will be deleted after 03-Apr-2020

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/71ead89f002cf2455d4d75c041781a30/raw/909b33bc2937ba936fb04c660212c26c0fe19c2e/_main) | :x: FAILURE
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/71ead89f002cf2455d4d75c041781a30/raw/70ad84d05f5114d98fc7fbb062008415ca149af7/r-vmb-ppc-jenkins_W0) | :x: FAILURE
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/71ead89f002cf2455d4d75c041781a30/raw/b2d8a2644384e221617521d749db93a441cf183e/r-vmb-ppc-jenkins_W1) | :x: FAILURE
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/71ead89f002cf2455d4d75c041781a30/raw/73fa28a89ad0ceded3c0734dd09e17e986c007cd/r-vmb-ppc-jenkins_W2) | :x: FAILURE
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/71ead89f002cf2455d4d75c041781a30/raw/e1e13f02cb51a2c99279d5929dee44dad3d0257a/r-vmb-ppc-jenkins_W3) | :x: FAILURE
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/71ead89f002cf2455d4d75c041781a30/raw/0f7c1bcaa06d63557e1e684b9bb7c7e3be2ca5b9/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/71ead89f002cf2455d4d75c041781a30/raw/6e802be2967b0ac8470ea6d2819c97df7aab650b/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/71ead89f002cf2455d4d75c041781a30/raw/1b49b23908243d7e64cd6026e86a1845cb4b8e92/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/71ead89f002cf2455d4d75c041781a30/raw/5ba189942541e54147beb7d8c0e69aacfd7a6b7b/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/71ead89f002cf2455d4d75c041781a30/raw/01b9c2854bb03982a9eeb9d38e91a0a2b15417c0/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/71ead89f002cf2455d4d75c041781a30/raw/5662862fca87150a10e984a851b5ab42dfc1c50e/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/71ead89f002cf2455d4d75c041781a30/raw/f13334d1e7c0d73deb68a8c3eb4bd1d62942786d/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/71ead89f002cf2455d4d75c041781a30/raw/c2f31d23a27304981101e1e68242ba716ca1369f/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/71ead89f002cf2455d4d75c041781a30/raw/a6d97da5545463c4f97a91fe8d383f5d8c67fd67/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/71ead89f002cf2455d4d75c041781a30/raw/2845a9b0ed677dbe6fe2050b488e6847a1dceba7/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/71ead89f002cf2455d4d75c041781a30/raw/ad3529760c448c6784d7e304414cfe7fe44634ba/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/71ead89f002cf2455d4d75c041781a30/raw/792c1b18afeb7f5864878501a62ff7161e3f02f7/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS

File: test/gtest/ucp/test_ucp_sockaddr.cc [hunk 1]
```diff
@@ -150,17 +150,22 @@ class test_ucp_sockaddr : public ucp_test {
...
+                saddrs.back().set_port(0); /* listen on any port then update */
             }
         }
+
         freeifaddrs(ifaddrs);
-        UCS_TEST_SKIP_R("No interface for testing");
+
+        if (saddrs.size() > 0) {
+            /* randomize iface */
+            m_test_addr = saddrs[ucs::rand() % saddrs.size()];
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_sockaddr.cc]: maybe just save pointers to "ifa" (and copy to m_test_addr before calling freeifaddrs)

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>FAILED</b> on 2 of 17 workers (click for details)</summary>

**Note:** the logs will be deleted after 06-Apr-2020

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/9c6b28d9b6940b71a36e29a8d68cad4d/raw/47ad790aa586b86996c221eb5743ec4fd7212131/_main) | :question: ABORTED
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/9c6b28d9b6940b71a36e29a8d68cad4d/raw/ad73a3cfe1667b3366057a9da6aa621f051845e4/hpc-arm-cavium-jenkins_W0) | :question: ABORTED
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/9c6b28d9b6940b71a36e29a8d68cad4d/raw/5188c8e72ba04fba6ccbd21a3fbc134a4c7be96e/hpc-arm-cavium-jenkins_W1) | :question: ABORTED
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/9c6b28d9b6940b71a36e29a8d68cad4d/raw/77b43ac5d79deb06dce0a2638a6529ef969b5596/hpc-arm-cavium-jenkins_W2) | :question: ABORTED
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/9c6b28d9b6940b71a36e29a8d68cad4d/raw/c87ba0a909134ba5462d59baedff9b99d4d5f185/hpc-arm-cavium-jenkins_W3) | :question: ABORTED
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/9c6b28d9b6940b71a36e29a8d68cad4d/raw/79b58b39935a8e3ccff9318c162f0d929f76ec3d/hpc-arm-hwi-jenkins_W0) | :question: ABORTED
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/9c6b28d9b6940b71a36e29a8d68cad4d/raw/22d662702f7c74772259805b10dd0dcdfcf08a2b/hpc-arm-hwi-jenkins_W1) | :question: ABORTED
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/9c6b28d9b6940b71a36e29a8d68cad4d/raw/78a2f5514e525dd417b4b6783ea4705767090c8f/hpc-arm-hwi-jenkins_W2) | :question: ABORTED
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/9c6b28d9b6940b71a36e29a8d68cad4d/raw/730abfd106508d2e589f92f1493d20ff2a08d3c7/hpc-arm-hwi-jenkins_W3) | :question: ABORTED
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/9c6b28d9b6940b71a36e29a8d68cad4d/raw/12117a1d0c8896b07af9b819a3ccd172bd21d90e/hpc-test-node-new_W1) | :question: ABORTED
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/9c6b28d9b6940b71a36e29a8d68cad4d/raw/a5e3b60838874b269e631d5f677b2403f65045dc/hpc-test-node-new_W2) | :question: ABORTED
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/9c6b28d9b6940b71a36e29a8d68cad4d/raw/c7633e07bada1ef9d6513422bdcfd63e8dd5d2d5/r-vmb-ppc-jenkins_W0) | :question: ABORTED
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/9c6b28d9b6940b71a36e29a8d68cad4d/raw/d72c117aefe99aa82e18c9d9dbff344c31dd1500/r-vmb-ppc-jenkins_W1) | :question: ABORTED
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/9c6b28d9b6940b71a36e29a8d68cad4d/raw/8c77c8afdb7cfec37745a9af1b73ead1eca8c379/r-vmb-ppc-jenkins_W2) | :question: ABORTED
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/9c6b28d9b6940b71a36e29a8d68cad4d/raw/6769712bc1f31c21896e8124b4cb95e315bd6ac3/r-vmb-ppc-jenkins_W3) | :question: ABORTED
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/9c6b28d9b6940b71a36e29a8d68cad4d/raw/fe59a8abec75ae5dd52542b4c5dd2205491b5f71/hpc-test-node-new_W0) | :x: FAILURE
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/9c6b28d9b6940b71a36e29a8d68cad4d/raw/1e396ebfc14d1b79ad1790fdb877635a81250350/hpc-test-node-new_W3) | :x: FAILURE

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>ABORTED</b> on 13 workers (click for details)</summary>

**Note:** the logs will be deleted after 06-Apr-2020

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/f07328a5fb9f534ffaf9026fea2ba628/raw/cacfc953202162d1f385bd3b3365a40a1452cb09/_main) | :question: ABORTED
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/f07328a5fb9f534ffaf9026fea2ba628/raw/18752f57afaf5fa6f02fb5d6bb388b51b57a7027/r-vmb-ppc-jenkins_W0) | :question: ABORTED
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/f07328a5fb9f534ffaf9026fea2ba628/raw/67a561932d606c809435c2b9a58df98cefb6e573/r-vmb-ppc-jenkins_W1) | :question: ABORTED
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/f07328a5fb9f534ffaf9026fea2ba628/raw/6f65fb38e04b574309056af88ef2673ce81be523/r-vmb-ppc-jenkins_W2) | :question: ABORTED
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/f07328a5fb9f534ffaf9026fea2ba628/raw/80ef0d2807d97f4c51c19f8c91806075629bd9ab/r-vmb-ppc-jenkins_W3) | :question: ABORTED
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/f07328a5fb9f534ffaf9026fea2ba628/raw/7287a9a96dc6955a631c08a0216e0f1ff301510a/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/f07328a5fb9f534ffaf9026fea2ba628/raw/085eab0c75b6245aeec4985ef70a54fb0e77cd91/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/f07328a5fb9f534ffaf9026fea2ba628/raw/34e41e376a23409b338e58018e77d7efa057f090/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/f07328a5fb9f534ffaf9026fea2ba628/raw/015f3aed145dab8a1edf00c174d5d386da0011a8/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/f07328a5fb9f534ffaf9026fea2ba628/raw/31938f32d190eb07a4ec0273875cceb6f8eda76b/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/f07328a5fb9f534ffaf9026fea2ba628/raw/1d29d577bb1178149659a0da2b248fdce2d17e91/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/f07328a5fb9f534ffaf9026fea2ba628/raw/f0fc0b349a381f294cf9932173ef168a4a291c78/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/f07328a5fb9f534ffaf9026fea2ba628/raw/c45de6e0348293d0cbc5f9516bce52c1e70089fb/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS

**Comment by evgeny-leksikov** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>FAILED</b> on 5 of 13 workers (click for details)</summary>

**Note:** the logs will be deleted after 07-Apr-2020

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/81f81edbe49b9f022354aba46bead457/raw/2ff2566a51cc92e62ca6153eccb986d2b1accf99/_main) | :x: FAILURE
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/81f81edbe49b9f022354aba46bead457/raw/9860d3590cccbc111e90c210b4ad20ad84229e01/r-vmb-ppc-jenkins_W0) | :x: FAILURE
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/81f81edbe49b9f022354aba46bead457/raw/70c7516289412151474ab5eb3b0e7a51a5942f2b/r-vmb-ppc-jenkins_W1) | :x: FAILURE
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/81f81edbe49b9f022354aba46bead457/raw/88853f1fc465d17aeb7f75abf824a194ff30a6d7/r-vmb-ppc-jenkins_W2) | :x: FAILURE
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/81f81edbe49b9f022354aba46bead457/raw/6ff85aa9c0b23d442de97d3133723b83d5a146e7/r-vmb-ppc-jenkins_W3) | :x: FAILURE
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/81f81edbe49b9f022354aba46bead457/raw/409301f832b3df98edaef4ae4c86f603262cfae9/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/81f81edbe49b9f022354aba46bead457/raw/579d63f1b6f614fb63b049048cc36ff343be05e0/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/81f81edbe49b9f022354aba46bead457/raw/ed564ee35a37500ac75feaea1fbd91412105b46a/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/81f81edbe49b9f022354aba46bead457/raw/0cc8edeaf7d47570abf5fa185e03481ace96fbb7/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/81f81edbe49b9f022354aba46bead457/raw/e43fd15663f0c0c58be3471286ea0b80f734a5d8/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/81f81edbe49b9f022354aba46bead457/raw/6f3f0f42c7d1041c03631893bf9508c5b7c1c7f7/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/81f81edbe49b9f022354aba46bead457/raw/dc7da523e6b05fae0bbd610324424726b28afe08/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/81f81edbe49b9f022354aba46bead457/raw/e4a2917b4a1142177962a65f11fabedcdd77d7d5/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS

**Comment by yosefe** [comment]
bot:pipe:retest

**Comment by yosefe** [comment]
bot:pipe:retest

**Comment by evgeny-leksikov** [comment]
@yosefe  this PR can't pass CI before https://github.com/openucx/ucx/pull/5002 is merged

**Comment by yosefe** [comment]
> @yosefe this PR can't pass CI before #5002 is merged

Got it, thanks

**Comment by evgeny-leksikov** [comment]
bot:pipe:retest

File: test/gtest/ucp/test_ucp_sockaddr.cc [hunk 2]
```diff
@@ -161,8 +161,17 @@ class test_ucp_sockaddr : public ucp_test {
...
 
         if (saddrs.size() > 0) {
-            /* randomize iface */
-            m_test_addr = saddrs[ucs::rand() % saddrs.size()];
+            static const std::string ud_or_dc_tls[] = { "ud", "ud_v", "ud_x",
+                                                        "dc", "dc_x", "ib" };
+
+            bool has_dc_or_ud = has_any_transport(
+                                    std::vector<std::string>(ud_or_dc_tls,
+                                                             ud_or_dc_tls + 6));
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_sockaddr.cc:169]: 6 -> ucs_array_size

File: test/gtest/ucp/test_ucp_sockaddr.cc [hunk 3]
```diff
@@ -161,8 +161,17 @@ class test_ucp_sockaddr : public ucp_test {
         freeifaddrs(ifaddrs);
 
         if (saddrs.size() > 0) {
-            /* randomize iface */
-            m_test_addr = saddrs[ucs::rand() % saddrs.size()];
+            static const std::string ud_or_dc_tls[] = { "ud", "ud_v", "ud_x",
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_sockaddr.cc:164]:
```
if (saddrs.empty()) {
  skip;
}

static const std::string ...
has_dc_or_ud = ...
int saddr_idx = has_dc_or_ud ? 0 : (ucs::rand() % saddrs.size());
m_test_addr   = saddrs[saddr_idx];
 

```

File: test/gtest/ucp/test_ucp_sockaddr.cc [hunk 4]
```diff
@@ -203,7 +217,7 @@ class test_ucp_sockaddr : public ucp_test {
     static void rtag_complete_cb(void *req, ucs_status_t status,
                                  ucp_tag_recv_info_t *info)
     {
-        EXPECT_UCS_OK(status);
+        ASSERT_UCS_OK_OR_CANCELED(status);
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_sockaddr.cc:220]: why needed?

Comment (evgeny-leksikov) [reply test/gtest/ucp/test_ucp_sockaddr.cc:220]: leftover

File: test/gtest/ucp/test_ucp_sockaddr.cc [hunk 5]
```diff
@@ -160,21 +160,22 @@ class test_ucp_sockaddr : public ucp_test {
...
+                                                    "dc", "dc_x", "ib" };
+
+        bool has_dc_or_ud = has_any_transport(
+                        std::vector<std::string>(ud_or_dc_tls,
+                                                 ud_or_dc_tls +
+                                                 ucs_array_size(ud_or_dc_tls)));
+
+        /* randomize iface excepting UD and DC transports which have a bug
+           on transport level, PR #5237 fixes ud/udx */
+        int saddr_idx = has_dc_or_ud ? 0 : ucs::rand() % saddrs.size();
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_sockaddr.cc:177]: use ( ) around `ucs::rand() % saddrs.size()`

File: test/gtest/ucp/test_ucp_sockaddr.cc [hunk 6]
```diff
@@ -150,17 +150,32 @@ class test_ucp_sockaddr : public ucp_test {
...
+        static const std::string ud_or_dc_tls[] = { "ud", "ud_v", "ud_x",
+                                                    "dc", "dc_x", "ib" };
+
+        bool has_dc_or_ud = has_any_transport(
+                        std::vector<std::string>(ud_or_dc_tls,
+                                                 ud_or_dc_tls +
+                                                 ucs_array_size(ud_or_dc_tls)));
+
+        /* randomize iface excepting UD and DC transports which have a bug
+           on transport level, PR #5237 fixes ud/udx */
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_sockaddr.cc:176]: FIXME select random interface, except for  UD and DC transports, which do not yet support having different gid_index for different UCT endpoints on same iface

File: test/gtest/ucp/test_ucp_sockaddr.cc [hunk 7]
```diff
@@ -203,7 +218,7 @@ class test_ucp_sockaddr : public ucp_test {
     static void rtag_complete_cb(void *req, ucs_status_t status,
                                  ucp_tag_recv_info_t *info)
     {
-        EXPECT_UCS_OK(status);
+        ASSERT_UCS_OK(status);
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_sockaddr.cc:221]: why needed?

File: test/gtest/ucp/test_ucp_sockaddr.cc [hunk 8]
```diff
@@ -901,7 +916,6 @@ class test_ucp_sockaddr_protocols : public test_ucp_sockaddr {
     virtual void init() {
         test_ucp_sockaddr::init();
         start_listener(cb_type());
-        client_ep_connect();
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_sockaddr.cc:904]: why needed?

File: test/gtest/ucp/test_ucp_sockaddr.cc [hunk 9]
```diff
@@ -957,25 +974,38 @@ class test_ucp_sockaddr_protocols : public test_ucp_sockaddr {
...
-        compare_buffers(send_buf, recv_buf);
+        if (wait(rreq, 0, true) == UCS_INPROGRESS) {
+            UCS_TEST_MESSAGE << "cancel not completed tag receive request";
+            ucp_request_cancel(receiver().worker(), rreq);
+            ucp_request_free(rreq);
+        }
+
+        if (num_failed_entities() == 0) {
+            compare_buffers(send_buf, recv_buf);
+        }
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_sockaddr.cc:986]: why needed?

File: test/gtest/ucp/test_ucp_sockaddr.cc [hunk 10]
```diff
@@ -957,25 +974,38 @@ class test_ucp_sockaddr_protocols : public test_ucp_sockaddr {
...
     }
 
     void wait_for_server_ep()
     {
         if (!test_ucp_sockaddr::wait_for_server_ep(false)) {
+            if (sender().get_err_num() > 0) {
+                UCS_TEST_SKIP_R("server endpoint is not created due to client connection failure");
+            }
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_sockaddr.cc:994]: why needed?

File: test/gtest/ucp/test_ucp_sockaddr.cc [hunk 11]
```diff
@@ -987,7 +1017,12 @@ class test_ucp_sockaddr_protocols : public test_ucp_sockaddr {
             sreq = ucp_stream_send_nb(sender().ep(), &send_buf[0], size,
                                    ucp_dt_make_contig(1), scomplete_cb, 0);
             short_progress_loop();
-            wait_for_server_ep();
+            try {
+                wait_for_server_ep();
+            } catch (const ucs::test_skip_exception &e) {
+                wait(sreq);
+                throw e;
+            }
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_sockaddr.cc:1025]: why needed?

**Comment by evgeny-leksikov** [comment]
> why need any changes in test_ucp_sockaddr_protocols ?

reverted, was wrongly added by mistake due to gid mismatch issue

**Pull request merged**